### PR TITLE
Support multi-word for project names and column names for project plugin

### LINKED ISSUE #12481: Project plugin incorrectly handles command with column name
**What happened**: Project plugin is not responding to commands with project name and column name as arguments. Example: https://github.com/kubeflow/community/issues/250

**What you expected to happen**: Project plugin should successfully add issue to project and column

/assign @taragu 
/sig testing

### PR DESCRIPTION
**Pull request by taragu**
Fixes #12481

/cc @stevekuznetsov 
/cc @cjwagner 
FYI @brahmaroutu

### CODE REVIEW

**Comment by mohammedzee1000** [comment]
Why not go all the way in being explicit and make it like 
/project set¦clear project="blah" column="blah"

**Comment by mohammedzee1000** [comment]
@taragu

**Comment by taragu** [comment]
@mohammedzee1000 thanks for the suggestion! I agree `/project set¦clear project="blah" column="blah"` is more straightforward. Will make the change

**Comment by cblecker** [comment]
is there a reason for the `set|clear` vs using the same command pattern as other commands `/[remove-]project`?

we don't use explicit variable pairs for any other command, so I'm reluctant to move that direction. the list of current command patterns is https://prow.k8s.io/command-help

**Comment by mohammedzee1000** [comment]
Well I guess we could go with remove pattern as well
Set : `/project project="blah" column="blah"`
Remove: `/remove-project project="blah"`
My recommendation is assuming we go route suggested by @taragu why not go all the way. 
See issue in description for discussion which lead here

**Comment by taragu** [comment]
I've updated this PR to follow the following command patterns:
Set : /project project="blah" column="blah"
Remove: /remove-project project="blah"
Can anyone review? @stevekuznetsov @cjwagner @cblecker

File: prow/plugins/project/project.go [hunk 1]
```diff
@@ -36,9 +36,12 @@ const (
 )
 
 var (
-	projectRegex              = regexp.MustCompile(`(?m)^/project\s+(.+?)(?:\s+(.+?)\s*)?$`)
+	projectRegex              = regexp.MustCompile(`(?m)^(?:\/(project)\s+(project=(?:'.*?'|".*?"))(?:\s+(column=(?:'.*?'|".*?"))\s*?$)?)|(?:\/(remove-project)\s+(project=(?:'.*?'|".*?"))\s*?$)`)
```
Comment (stevekuznetsov) [inline prow/plugins/project/project.go]: If we don't care to enforce matching quotation marks (which seems fine to me?) we can simplify this a lot:

https://play.golang.org/p/MWNQIaywL5E

Comment (taragu) [reply prow/plugins/project/project.go]: @stevekuznetsov yeah that does seem a lot simpler. I think it's fine to allow a command like `/project project='0.0.0" column="To do'`. Thanks for the suggestion!

**Pull request closed without merging**